vividus-plugin-azure-storage-account/src/test/java/org/vividus/azure/storage/blob/BlobStorageStepsTests.java

```diff
@@ -180,13 +194,133 @@ void shouldListFilteredBlobs()
+                arguments(
+                        Optional.of(StringComparisonRule.MATCHES), Optional.of(".*"),
+                        Optional.empty(), Optional.empty(), List.of(FIRST, SECOND, THIRD)),
+                arguments(
+                        Optional.empty(), Optional.empty(),
+                        Optional.of(DATASETS), Optional.empty(), List.of(FIRST, SECOND, THIRD)),
+                arguments(
+                        Optional.of(StringComparisonRule.MATCHES), Optional.of("null"),
+                        Optional.of(DATASETS), Optional.of(1L), List.of())
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("matchingParametersToBlobs")
+    @SuppressWarnings("unchecked")
+    void shouldListFilteredBlobs(Optional<StringComparisonRule> rule, Optional<String> name, Optional<String> prefix,
+                                 Optional<Long> limit, List<String> blobs)
+    {
+        BlobFilter filter = new BlobFilter();
+        filter.setBlobNameFilterRule(rule);
+        filter.setBlobNameFilterValue(name);
+        filter.setBlobNamePrefix(prefix);
+        filter.setResultsLimit(limit);
+
+        runWithClient((steps, client) ->
+        {
+            BlobItem first = blobItem(FIRST);
+            BlobItem second = blobItem(SECOND);
+            BlobItem third = blobItem(THIRD);
+
+            limit.ifPresentOrElse(
+                    l -> {
+                        PagedIterable<BlobItem> pagedResponse = mock(PagedIterable.class);
+                        PagedResponse<BlobItem> page1 = mock(PagedResponse.class);
+                        PagedResponse<BlobItem> page2 = mock(PagedResponse.class);
+                        when(pagedResponse.iterableByPage())
+                                .thenReturn(List.of(page1, page2, mock(PagedResponse.class)));
+                        when(page1.getValue()).thenReturn(List.of(first, second));
+                        when(page2.getValue()).thenReturn(List.of(third));
+
+                        prefix.ifPresentOrElse(
+                                p -> when(client.listBlobsByHierarchy(eq("/"), any(ListBlobsOptions.class),
+                                        eq(null))).thenReturn(pagedResponse),
+                                () -> when(client.listBlobs(any(ListBlobsOptions.class), eq(null)))
+                                        .thenReturn(pagedResponse)
+                        );
+                    },
+                    () ->  {
+                        PagedIterable<BlobItem> iterable = mock(PagedIterable.class);
+                        when(iterable.stream()).thenReturn(Stream.of(first, second, third));
+                        prefix.ifPresentOrElse(
+                                p -> when(client.listBlobsByHierarchy(eq(p))).thenReturn(iterable),
+                                () -> when(client.listBlobs()).thenReturn(iterable)
+                        );
+                    }
+            );
```

Comment:
unit tests shouldn't contain so many `if`-logic (aka code branches), in the ideal state there should be no branches at all